src/dreamer/specialists.py

```
@@ -0,0 +1,434 @@
+        # Log metrics
+        duration_ms = (time.perf_counter() - start_time) * 1000
+        accumulate_metric(task_name, "total_duration", duration_ms, "ms")
+        accumulate_metric(
+            task_name, "tool_calls", len(response.tool_calls_made), "count"
+        )
+        accumulate_metric(task_name, "input_tokens", response.input_tokens, "count")
+        accumulate_metric(task_name, "output_tokens", response.output_tokens, "count")
+
+        logger.info(
+            f"{self.name}: Completed in {duration_ms:.0f}ms, "
+            + f"{len(response.tool_calls_made)} tool calls, "
```

Ruff (0.14.10)

Explicitly concatenated string should be implicitly concatenated

Remove redundant '+' operator to implicitly concatenate

(ISC003)